utilities-core/src/main/java/uk/gov/justice/services/common/converter/jackson/integerenum/EnumObjectFinder.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.justice.services.common.converter.jackson.integerenum;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+
+public class EnumObjectFinder {
+
+    public Optional<Object> findEnumObject(final Field[] declaredFields, final Object enumObject, final int enumValue) {
+        for (final Field field : declaredFields) {
+            if (Integer.class.isAssignableFrom(field.getType())) {
+                try {
+                    field.setAccessible(true);
+                    final int fieldValue = (int) field.get(enumObject);
+                    if (fieldValue == enumValue) {
+                        return of(enumObject);
+                    }
+                } catch (final IllegalAccessException e) {
```

Comment:
Maybe change the 'e' of the Exception to 'doNothing'  or 'ignored' then you won't need the comment